test/lib/test_ocaml_delimiter.ml

```diff
@@ -0,0 +1,33 @@
+open Mdx.Ocaml_delimiter
+open Result
+
+let test_parse =
+  let make_test ~line ~expected =
+    let test_name = Printf.sprintf "parse: %S" line in
+    let test_fun () =
+      let actual = parse line in
+      Alcotest.(check (result (option Testable.ocaml_delimiter) Testable.msg))
+        test_name expected actual
+    in
+    (test_name, `Quick, test_fun)
+  in
+  [
+    make_test ~line:"" ~expected:(Ok None);
+    make_test ~line:"foo" ~expected:(Ok None);
+    make_test ~line:"    (* $MDX part-begin=bar *)    "
+      ~expected:
+        (Ok (Some (Part_begin (Cmt, { indent = "    "; payload = "bar" }))));
+    (* Unclosed comments are caught by syntax highlighting, it is okay to
+       silently ignore them and not print a warning here. *)
+    make_test ~line:"(* $MDX part-begin=bar     " ~expected:(Ok None);
+    make_test ~line:"   (* $MDX part-end   *)   " ~expected:(Ok (Some Part_end));
+    make_test ~line:"    [@@@part \"foobar\"]    "
+      ~expected:
+        (Ok (Some (Part_begin (Attr, { indent = "    "; payload = "foobar" }))));
+    make_test ~line:"(* $MDX foo *)"
+      ~expected:
+        (Error (`Msg "'(* $MDX foo *)' is not a valid ocaml delimiter for mdx"));
+    make_test ~line:"[@@@foo \"bar\"]" ~expected:(Ok None);
```

Comment:
Might be good to add a test case for `part-end=...`, I feel like this is a confusion that might happen quite often, we might even want to have a specific error for this one.

Reply:
Good idea, done.